src/dispatch.c

```diff
@@ -82,3 +83,452 @@ natsStatus natsSub_enqueueUserMessage(natsSubscription *sub, natsMsg *msg)
     return NATS_OK;
 }
 
+static inline void _removeFromDispatchQueue(natsDispatcher *d, natsMsg *msg)
```

Comment:
Don't know if this was done before stopping the "format on save" with the clang-formatter, but if you want/could, I would have the return type in one line and the the function name in the next, to be consistent with (most of) the rest of the library.